packages/peregrine/lib/hooks/useCustomerWishlistSkus/useCustomerWishlistSkus.js

```diff
@@ -0,0 +1,69 @@
+import { useState } from 'react';
+import { useQuery } from '@apollo/client';
+import { useUserContext } from '../../context/user';
+import mergeOperations from '../../util/shallowMerge';
+
+import defaultOperations from './customerWishlist.gql.ee';
+
+/**
+ * A hook that queries for products in a customer's wishlists and maintains a
+ * list of skus in a local cache entry.
+ *
+ * @param {DocumentNode} props.operations operations used to prepare the local customer wishlist array
+ * @returns {undefined}
+ */
+export const useCustomerWishlistSkus = (props = {}) => {
+    const operations = mergeOperations(defaultOperations, props.operations);
+    const [{ isSignedIn }] = useUserContext();
+
+    const [currentPage, setCurrentPage] = useState(1);
+
+    const {
+        client,
+        data: { customerWishlistProducts }
+    } = useQuery(operations.getProductsInWishlistsQuery);
+
+    useQuery(operations.getWishlistItemsQuery, {
```

Comment:
It's possible that this hook is called multiple times on a single page by different components, for example if a carousel is rendered on a category page.

In that event, we _should_ be safe to assume that the `onCompleted` callbacks are invoked synchronously, and would result in the same list of skus regardless of order.

Reply:
I have checked on the page with multiple instances of PageBuilder Carousels and it seems no problems